### PR TITLE
EZP-31097: Introduced strict types for URLService

### PR DESCRIPTION
**Pull request by adamwojs**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31097](https://jira.ez.no/browse/EZP-31097)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `master`
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | yes

This PR introduces strict types for the` \eZ\Publish\API\Repository\URLService`

**TODO**:
- [X] Implement feature / fix a bug.
- [X] Implement tests.
- [X] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/Repository/URLService.php [hunk 1]
```diff
@@ -245,9 +248,10 @@ protected function buildUpdateStruct(URL $url, URLUpdateStruct $data)
      * @param string $url
      *
      * @return bool
+     *
      * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
      */
-    protected function isUnique($id, $url)
+    protected function isUnique($id, $url): bool
```
Comment (ViniTou) [inline eZ/Publish/Core/Repository/URLService.php]:
```suggestion
    protected function isUnique(int $id, string $url): bool
```

File: eZ/Publish/Core/Repository/URLService.php [hunk 2]
```diff
@@ -256,7 +260,7 @@ protected function isUnique($id, $url)
         }
     }
 
-    private function createDateTime($timestamp)
+    private function createDateTime($timestamp): ?DateTimeInterface
```
Comment (ViniTou) [inline eZ/Publish/Core/Repository/URLService.php]:
```suggestion
    private function createDateTime(int $timestamp): ?DateTimeInterface
```

File: eZ/Publish/Core/Repository/URLService.php (hunk 1 above)
Comment (adamwojs) [reply eZ/Publish/Core/Repository/URLService.php]: Fixed in https://github.com/ezsystems/ezpublish-kernel/pull/2854/commits/1719f1cf4a628cce40bbf098c2172c7ceae4d1f9

File: eZ/Publish/Core/Repository/URLService.php (hunk 2 above)
Comment (adamwojs) [reply eZ/Publish/Core/Repository/URLService.php]: Fixed in https://github.com/ezsystems/ezpublish-kernel/pull/2854/commits/1719f1cf4a628cce40bbf098c2172c7ceae4d1f9

**Comment by adamwojs** [comment]
PR rebased in order to fix code style issues

File: eZ/Publish/API/Repository/URLService.php [hunk 3]
```diff
@@ -20,17 +24,16 @@ interface URLService
      *
      * @return \eZ\Publish\API\Repository\Values\URL\URLUpdateStruct
      */
-    public function createUpdateStruct();
+    public function createUpdateStruct(): URLUpdateStruct;
 
     /**
      * Find URLs.
      *
-     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/URLService.php]: Should we delete this?

File: eZ/Publish/API/Repository/URLService.php [hunk 4]
```diff
@@ -40,42 +43,46 @@ public function findUrls(URLQuery $query);
...
-     * @param int $id ID of URL
      * @return \eZ\Publish\API\Repository\Values\URL\URL
      */
-    public function loadById($id);
+    public function loadById(int $id): URL;
 
     /**
      * Load single URL (by URL).
      *
+     * @param string $url url
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/URLService.php]:
```suggestion
     * @param string $url URL
```

**Comment by adamwojs** [comment]
PR updated according to @mikadamczyk code review suggestions.

File: eZ/Publish/API/Repository/URLService.php (hunk 3 above)
Comment (adamwojs) [reply eZ/Publish/API/Repository/URLService.php]: Good catch @mikadamczyk. Restored in https://github.com/ezsystems/ezpublish-kernel/pull/2854/commits/3ec650a651309cd5e0500b11bb3fa3e760902084

**Pull request merged**